src/App.jsx

```diff
@@ -1,14 +1,58 @@
 import './App.css';
+import messages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.jsx';
+import ColorChoice from './components/ColorChoice.jsx';
 
 const App = () => {
+  const [messageData, setMessage] = useState(messages);
+  const [mainSenderColor, setMainSenderColor] = useState('black');
+  const [remoteSenderColor, setRemoteSenderColor] = useState('black');
+  const MainSender = 'Vladimir';
+  const senderNames = [...new Set(messageData.map(message => message.sender))];
+  const remoteSender = senderNames.filter(sender => sender!== MainSender);
+
+  const likeMessage = (id) => {
+    setMessage(messageData => {
+      return messageData.map(message => {
+        if (message.id == id){
+          return {...message, liked: !message.liked};
+        }else {
```

Comment:
Nitpick: missing white space after }
```suggestion
        } else {
```